pkg/cloud/ovirt/machine/actuator.go

```diff
@@ -278,9 +280,19 @@ func (actuator *OvirtActuator) patchMachine(machine *machinev1.Machine, instance
 
 func (actuator *OvirtActuator) reconcileNetwork(machine *machinev1.Machine, instance *clients.Instance) error {
 	switch instance.MustStatus() {
-	// expect IP addresses only on those statues.
-	case ovirtsdk.VMSTATUS_UP, ovirtsdk.VMSTATUS_MIGRATING, ovirtsdk.VMSTATUS_POWERING_UP:
+	// expect IP addresses only on those statuses.
+	// in those statuses we 'll try reconciling
+	case ovirtsdk.VMSTATUS_UP, ovirtsdk.VMSTATUS_MIGRATING:
 		break
+
+	// return error if vm is rebooting this will force retry reconciling.
+	// unless we retry reconciling this transient state will remain forever
+	// there is no event generated that will trigger this.  BZ1854787
+	case ovirtsdk.VMSTATUS_REBOOT_IN_PROGRESS, ovirtsdk.VMSTATUS_POWERING_UP:
+		return fmt.Errorf("Aborting reconciliation while VM %s  state is %s", instance.MustName(), instance.MustStatus())
+
+	// skip network reconciling ,this will mark reconciling as success , it wont be retried.
+	// rely on events for next reconciling attempt.
 	default:
```

Comment:
Because...what's the point of skipping the resolution? If we get called it means something must have happened to kubelet. If it is a VM restart you can miss intermediate states and then you'd have stale data. If it's anything else then does it really matter if we "update" with the same IPs? I guess not (logs seem to indicate that nothing happens in that case)

so isn't it easier to just return error in transient cases (reboot, powering up, ...) and fall through any other time?
and then just keep returning error just in case, until you actually get some IP from net.LookupIP.
or maybe really just rely on that....
always lookup the hostname, if there's no error(or rather if the IP list is not empty?) just assign to machine.Status.Addresses and return nil. Otherwise abort so you trigger a retry (for handling transient states)

Reply:
we must take into consideration that there are other reconcilers (that are not network) if we return error every time they won't be triggered for example https://github.com/openshift/cluster-api-provider-ovirt/blob/0d50dd86aa58571757d528275409de766d0d43ff/pkg/cloud/ovirt/machine/actuator.go#L321

Reply:
the question is if they are any good if you don't have the node's IP. 
There should be some mechanism so this doesn't get stuck in infinite retry...the current situation when it's stuck in infinite stale data is not any better either.
I'd say it still best to return error in transient states but in any normal state (which is really just Up and Migrating) we should be able to get the IP and if not then something is really wrong and not running correctly...and in that case we should stop and set an error. Maybe with handleMachineError?

Reply:
okay , agreed  lets s simplifying the code , updated.